### ISSUE TITLE
Find a home for new module documentation

### ISSUE THREAD

**Issue by geerlingguy** [ansible-collections/community.kubernetes#31]
##### SUMMARY

We have two new modules in this collection as of today:

  - k8s_exec: https://github.com/ansible-collections/kubernetes/pull/14
  - k8s_log: https://github.com/ansible-collections/kubernetes/pull/16

These modules have appropriate documentation with examples, but currently there is nowhere public that the docs are displayed. For the other existing modules, which were included in Ansible 2.9 and earlier, we link in the README file to the Ansible module documentation.

For this issue, we need to figure out:

  - [ ] Will documentation be displayed in the collection UI in Ansible Galaxy?
  - [ ] When will docs in the main Ansible documentation go away? (If ever)
  - [ ] Should we move the links in the README to somewhere else? (right now k8s_exec and k8s_log link to their actual module python files, which are not very readable)

##### ISSUE TYPE

- Documentation

##### ADDITIONAL INFORMATION

Related issues:

  - https://github.com/ansible/galaxy/issues/2225

**Comment by gundalow** [comment]
Once https://github.com/ansible/ansible/pull/59761 is done you should get this for free
Something along the lines of docs.ansible.com/collections/community/kubernetes/modules/k8s_exec.html

Once that's done you could update README to point to these URLs

**Comment by geerlingguy** [comment]
@gundalow - From some of my quick reading of that PR and other related issues, it seems like it might be basing things on BOTMETA and some of the migration scripts... which would be great for existing stuff in 2.9 that's being migrated—but what about the new modules?

Will we need to file PRs against ansible/ansible any time we want to add a new module and get its docs added to the docsite? Or will docs.ansible.com scan all collections in Ansible Galaxy to generate the docs?

**Comment by gundalow** [comment]
I *think* we will generate all documentation for any collection listed in BOTMETA, not limited to only files listed in BOTMETA, though let me go and check.

**Comment by abadger** [comment]
We will be building the docs for a whole collection.  So if it is a new module in a collection that is already going to be a part of ansible-2.10, it will end up on docs.ansible.com.

@geerlingguy Is that good enough for your uses?  We can close this if so.

**Comment by geerlingguy** [comment]
@abadger - I'd still like to keep this open until it happens. There are already some modules that currently have no docs page available anywhere, and for now, we may add a temporary `docs` folder in this repo that's generated from the modules in the repo.

**Comment by geerlingguy** [comment]
See also: https://github.com/ansible-network/collection_prep/blob/master/README.md#doc-generator

**Comment by geerlingguy** [comment]
My laziness could pay off... the testing docs site has a [preview](http://docs.testing.ansible.com/ansible/2.10/collections/community/kubernetes/index.html#plugins-in-community-kubernetes) of what docs could look like in Ansible 2.10.

Still would be nice to have up-to-date rendered docs on Galaxy though.

**Comment by geerlingguy** [comment]
In https://github.com/ansible-collections/community.kubernetes/pull/172 I updated the links in the README to go to the Ansible 2.10 docs site, where there's an index of all the plugins in this collection: https://docs.ansible.com/ansible/2.10/collections/community/kubernetes/

It would be nice if Galaxy could also render documentation (follow https://github.com/ansible/galaxy/issues/2225 for that), but I think the stable Ansible docs site is adequate for our needs for now, and this issue can be closed.
